sklearn/ensemble/gradient_boosting_cv.py

```diff
@@ -0,0 +1,825 @@
+
+        self.grid_scores_ = grid_scores
+        self.best_score_tuple_ = best_tuple
+        self.best_params_ = best_params
+
+        if self.refit:
+            self.best_params_['random_state'] = self.random_state
+            gb = self._estimator_class(**self.best_params_)
+            gb.fit(X, y)
+            self.best_estimator_ = gb
+
+        return self
+
+    def predict(self, X):
+        """Call ``predict`` on the best estimator.
+
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The input samples.
+        Returns
+        -------
+        y: array of shape = ["n_samples]
+            The predicted values.
+        """
+        return self.best_estimator_.predict(X)
+
+    def predict_proba(self, X):
+        """Call ``predict_proba`` on the best estimator.
+
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The input samples.
+
+        Raises
+        ------
+        AttributeError
+            If the ``loss`` does not support probabilities.
+
+        Returns
+        -------
+        p : array of shape = [n_samples]
+            The class probabilities of the input samples. The order of the
+            classes corresponds to that in the attribute ``classes_``.
+        """
+        return self.best_estimator_.predict_proba(X)
+
+    def predict_log_proba(self, X):
+        """Call ``predict_log_proba`` on the best estimator.
+
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The input samples.
+        Raises
+        ------
+
+        AttributeError
+            If the ``loss`` does not support probabilities.
+
+        Returns
+        -------
+        p : array of shape = [n_samples]
+            The class probabilities of the input samples. The order of the
+            classes corresponds to that in the attribute ``classes_``.
+        """
+        return self.best_estimator_.predict_log_proba(X)
+
+    def _get_params(self):
+        "Return parametrs to iterate over for underlying gradient boosting"
+
+        params = {
+            'loss': self.loss,
+            'learning_rate': self.learning_rate,
+            'subsample': self.subsample,
+            'min_samples_split': self.min_samples_split,
+            'min_samples_leaf': self.min_samples_leaf,
+            'min_weight_fraction_leaf': self.min_weight_fraction_leaf,
+            'max_depth': self.max_depth,
+            'init': self.init,
+            'max_features': self.max_features,
+            'max_leaf_nodes': self.max_leaf_nodes,
+            'presort': self.presort,
+        }
+
+        # Pre processing to ensure every parameter is a list
+        # which ``ParameterGrid`` can iterate over
+        for key, value in params.items():
+            params[key] = np.atleast_1d(value)
+
+        return params
+
+
+class GradientBoostingRegressorCV(BaseEstimator):
```

Comment:
Indeed there was some discussion on subclassing `GridSearchCV`, but I think `_BaseGradientBoostingCV` would be better to avoid unintended side effects... I'll do this and ping you. Thanks again for the review!